src/cache.rs

```diff
@@ -504,4 +517,30 @@ mod tests {
 
         assert_eq!(&buf, b"Hello\n");
     }
+
+    macro_rules! https_client_tests {
+        ($($name:ident: $backend:expr),* $(,)?) => {
+            $(
+                #[test]
+                fn $name() {
+                    let dir = tempfile::tempdir().unwrap();
+
+                    let _ = Cache::build_client(&Cache {
+                        cache_dir: dir.into_path(),
+                        enable_styles: false,
+                        tls_backend: $backend,
+                    });
```

Comment:
We are not checking whether the returned `Result` of `build_client` is `Ok` here - let's do so

Additionally, I think we can make `build_client` not take the whole `Cache` but just the `TlsBackend` parameter. Then we won't need the tempdir and we can then remove the macro too and write out the three tests